File: mau/57283.cpp
#include <iostream>
using namespace std;
struct DIEM {
    int x, y;
};
DIEM A[1001];
int n;
bool xeptruoc(DIEM a, DIEM b) {
    int ka = a.x * a.x + a.y * a.y;
    int kb = b.x * b.x + b.y * b.y;
    if (ka < kb || (ka==kb && a.x>b.x)) 
        return true;
    else return false;
}
int main()
{
    cin>>n;
    int i,j;
    for (i=1; i<=n; i++) 
        cin>>A[i].x>>A[i].y;
    for (i=1; i<n; i++) 
        for (j=i+1; j<=n; j++)
            if (!xeptruoc(A[i],A[j])) 
                swap(A[i], A[j]);
    for (i=1; i<=n; i++)
        cout<<A[i].x<<" "<<A[i].y<<endl;
    return 0;
}

File: mau/57283.pas
type DIEM = record
    x, y : longint;
end;
function xeptruoc(a, b : DIEM) : boolean;
var ka, kb : integer;
begin
    ka:= sqr(a.x) + sqr(a.y);
    kb:= sqr(b.x) + sqr(b.y);
    if (ka<kb)or((ka=kb)and(a.x>b.x)) then exit(true)
    else exit(false)
end;
var n, i, j : integer;
    A : array[1..1000] of DIEM;
    tmp : DIEM;
BEGIN
    readln(n);
    for i:= 1 to n do 
        readln(A[i].x, A[i].y);
    for i:= 1 to n-1 do
        for j:= i+1 to n do
            if not xeptruoc(A[i], A[j]) then 
            begin
                tmp:= A[i];
                A[i]:= A[j];
                A[j]:= tmp;
            end;
    for i:= 1 to n do 
        writeln(A[i].x,' ',A[i].y);
END.
